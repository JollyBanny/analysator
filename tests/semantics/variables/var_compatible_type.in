var a : integer = 1;
var b : string = 'some string';
var c : char = 'c';
var d : double = 1.1;
var d1 : double = 11;
var f : boolean = false;
var t : boolean = true;

begin
end.