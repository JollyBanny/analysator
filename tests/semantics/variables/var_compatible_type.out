program
└──── program block
      ├──── VAR
      │     └──── var declaration
      │           ├──── a
      │           ├──── integer
      │           └──── 1
      ├──── VAR
      │     └──── var declaration
      │           ├──── b
      │           ├──── STRING
      │           └──── 'some string'
      ├──── VAR
      │     └──── var declaration
      │           ├──── c
      │           ├──── char
      │           └──── 'c'
      ├──── VAR
      │     └──── var declaration
      │           ├──── d
      │           ├──── double
      │           └──── 1.1
      ├──── VAR
      │     └──── var declaration
      │           ├──── d1
      │           ├──── double
      │           └──── cast to double
      │                 └──── 11
      ├──── VAR
      │     └──── var declaration
      │           ├──── f
      │           ├──── boolean
      │           └──── FALSE
      ├──── VAR
      │     └──── var declaration
      │           ├──── t
      │           ├──── boolean
      │           └──── TRUE
      └──── Compound statement
            └──── Empty statement

------------------------------------------------------------
|                         builtins                          |
------------------------------------------------------------
|            Type             |           integer           |
------------------------------------------------------------
|            Type             |           double            |
------------------------------------------------------------
|            Type             |            char             |
------------------------------------------------------------
|            Type             |           string            |
------------------------------------------------------------
|            Type             |           boolean           |
------------------------------------------------------------
------------------------------------------------------------
|                          global                           |
------------------------------------------------------------
|        Var        |         a         |      integer      |
------------------------------------------------------------
|        Var        |         b         |      string       |
------------------------------------------------------------
|        Var        |         c         |       char        |
------------------------------------------------------------
|        Var        |         d         |      double       |
------------------------------------------------------------
|        Var        |        d1         |      double       |
------------------------------------------------------------
|        Var        |         f         |      boolean      |
------------------------------------------------------------
|        Var        |         t         |      boolean      |
------------------------------------------------------------
